src/py_sonar_scanner/configuration.py

```diff
@@ -50,8 +50,10 @@ def setup(self) -> None:
         """This is executed when run from the command line"""
         self._read_wrapper_arguments()
         ApplicationLogger.set_debug(self.is_debug())
-        self.scan_arguments = sys.argv[1:]
-        self.scan_arguments.extend(self._read_toml_args())
+        self.scan_arguments = self._read_toml_args()
+        # If duplicate properties are provided, newer values will override older ones.
+        # We therefore read CLI arguments last so that they have priority over toml configuration.
+        self.scan_arguments.extend(sys.argv[1:])
```

Comment:
I would like to mention something I meant to mention in Maksim's PR, but that I didn't have a chance to. It is a bit unrelated to this PR, however the fix is small, therefore I think it might still be worth to incorporate this.

Currently, we have that `sys.argv[1:]` is being parsed in `_read_wrapper_arguments`, and relevant argument are fetched there. Therefore I don't really see why `self.scan_arguments` needs to be extended by the entire `sys.argv[1:]`, because I think that we would be passing irrelevant argument to the scanner (such as the `.toml`, or the `debug` property, which I believe are only relevant for the wrapper). 

Therefore as a refactor I suggest making use of the fact that `ArgumentParser.parse_known_args`'s second return value is all the arguments that couldn't be parsed, and therefore the only ones necessary to be passed down to the cli. I suggest the following:
```java
    def setup(self) -> None:
        """This is executed when run from the command line"""
        self. wrapper_arguments, remaining_args = self._read_wrapper_arguments()
        ApplicationLogger.set_debug(self.is_debug())
        self.scan_arguments = self._read_toml_args()
        # If duplicate properties are provided, newer values will override older ones.
        # We therefore read CLI arguments last so that they have priority over toml configuration.
        self.scan_arguments.extend(remaining_args[1:])

    def _read_wrapper_arguments(self):
        argument_parser = argparse.ArgumentParser()
        argument_parser.add_argument("-toml.path", "-Dtoml.path", "--toml.path", dest="toml_path")
        argument_parser.add_argument("-project.home", "-Dproject.home", "--project.home", dest="project_home")
        argument_parser.add_argument("-X", action="store_true", dest="debug")
        return argument_parser.parse_known_args(args=sys.argv[1:])
```

Reply:
This is a very good point. 

However we should remember some arguments are actually used by both the wrapper and the `sonar-scanner-cli`, for instance the `project.home` or the `-X` flag. Also we should make sure that we don't risk filtering out properties that may one day become relevant for `sonar-scanner-cli` (for instance, what if `toml` files were eventually supported natively by the `sonar-scanner-cli`?).


As discussed, I created [PYSCAN-43](https://sonarsource.atlassian.net/jira/software/c/projects/PYSCAN/issues/PYSCAN-43) to ensure we are at least intentional on this point.